web-marketplace/src/components/templates/ProjectDetails/ProjectDetails.DataStream.Post.tsx

```diff
@@ -0,0 +1,137 @@
+  ApolloClient,
+  NormalizedCacheObject,
+  useApolloClient,
+} from '@apollo/client';
+import TimelineConnector from '@mui/lab/TimelineConnector';
+import TimelineContent from '@mui/lab/TimelineContent';
+import TimelineItem from '@mui/lab/TimelineItem';
+import TimelineSeparator from '@mui/lab/TimelineSeparator';
+import { useQuery } from '@tanstack/react-query';
+import { useSetAtom } from 'jotai';
+
+import PostCard from 'web-components/src/components/cards/PostCard/PostCard';
+import { BubbleIcon } from 'web-components/src/components/icons/BubbleIcon';
+import { LockIcon } from 'web-components/src/components/icons/LockIcon';
+import { isImage } from 'web-components/src/components/inputs/new/FileDrop/FileDrop.utils';
+import { COPY_SUCCESS } from 'web-components/src/components/organisms/ProfileHeader/ProfileHeader.constants';
+import { Subtitle } from 'web-components/src/components/typography';
+import copyTextToClipboard from 'web-components/src/utils/copy';
+
+import { bannerTextAtom } from 'lib/atoms/banner.atoms';
+import { Post } from 'lib/queries/react-query/registry-server/getPostsQuery/getPostsQuery.types';
+import { getAccountByIdQuery } from 'lib/queries/react-query/registry-server/graphql/getAccountByIdQuery/getAccountByIdQuery';
+
+import { useAttestEvents } from 'pages/Post/hooks/useAttestEvents';
+import { ADMIN, POST_IS_PRIVATE } from 'pages/Post/Post.constants';
+import { DEFAULT_NAME } from 'pages/ProfileEdit/ProfileEdit.constants';
+import { getDefaultAvatar } from 'pages/ProfileEdit/ProfileEdit.utils';
+
+import { API_URI, IMAGE_STORAGE_BASE_URL } from './ProjectDetails.config';
+import {
+  FILES_ARE_PRIVATE,
+  LOCATIONS_ARE_PRIVATE,
+  PRIVATE_POST,
+} from './ProjectDetails.constant';
+
+type Props = {
+  post: Post;
+  index: number;
+  postsLength: number;
+  isAdmin: boolean;
+  adminAccountId?: string | null;
+};
+export const DataStreamPost = ({
+  post,
+  index,
+  postsLength,
+  isAdmin,
+  adminAccountId,
+}: Props) => {
+  const graphqlClient =
+    useApolloClient() as ApolloClient<NormalizedCacheObject>;
+  const setBannerText = useSetAtom(bannerTextAtom);
+
+  const { iri, createdAt } = post;
+
+  const { data: creatorAccountData } = useQuery(
+    getAccountByIdQuery({
+      client: graphqlClient,
+      id: post.creatorAccountId,
+      enabled: !!graphqlClient,
+    }),
+  );
+  const creatorAccount = creatorAccountData?.accountById;
+  const creatorIsAdmin = creatorAccount?.id === adminAccountId;
+
+  const { events } = useAttestEvents({
+    iri,
+    createdAt,
+    creatorAccount,
+    creatorIsAdmin,
+  });
+
+  const mimeTypes = post.filesMimeTypes;
+  let firstImageIri: string | undefined;
+  let firstImageUrl: string | undefined;
+  if (mimeTypes && post.filesUrls) {
+    firstImageIri = Object.keys(mimeTypes).find(key => isImage(mimeTypes[key]));
+    if (firstImageIri) {
+      firstImageUrl = post.filesUrls[firstImageIri];
+    }
+  }
+
+  return (
+    <TimelineItem key={post.iri}>
```

Comment:
I can see that the `<DataStreamPost>` component  is already receiving a `key` in `<DataStream>` when the posts are mapped. Could you please clarify the purpose of the `key` in this case?

Reply:
good catch, I believe it might just be I forgot to remove it because I've copied pasted `<TimelineItem ...>` from somewhere else where it's used so I'll remove it from here, thanks!